js/NavMainComponent.js

```diff
@@ -2,104 +2,249 @@
+    var component = this;
+
+    if (component.$html.find('[data-nav="' + target + '"]').length < 1) {
+        component.closeNavs();
+    }
 
-    component.$html.find('.nav-list.is-active').removeClass('is-active');
-    component.$html.find('[data-nav="' + target + '"]')
-        .addClass('is-active');
+    component.$html.find('[href="' + target + '"]').addClass('is-active');
 };
 
-NavMainComponent.prototype.changeActiveSecondaryNavLink = function(target) {
+NavMainComponent.prototype.switchQuaternaryNav = function (target) {
+    var component = this;
+
+    // If category item has encapsulated options and if it is same-page link then open the proper menu
+    if (target.indexOf('#') > -1) {
+        component.$navQuaternary.addClass('is-open');
+    }
 
+    component.$html.find('[data-nav="' + target + '"]').addClass('is-active');
+};
+
+NavMainComponent.prototype.changeActiveSecondaryNavLink = function (target) {
     var component = this;
 
-    component.$html.find('.nav-secondary .nav-item.is-active').removeClass('is-active');
+    component.$navSecondary.find('.nav-item.is-active').removeClass('is-active');
     component.$html.find('[href="' + target + '"]').addClass('is-active');
 };
 
-NavMainComponent.prototype.closeNavs = function() {
+NavMainComponent.prototype.changeActiveQuaternaryNavLink = function (target) {
+    var component = this;
+
+    component.$html.find('[href="' + target + '"]').addClass('is-active');
+};
 
+NavMainComponent.prototype.closeNavs = function () {
     var component = this;
 
-    component.$navMain.removeClass('is-open');
+    if (component.$navSecondary.hasClass('is-open')) {
+        component.$navSecondary.removeClass('is-open');
+    }
 
-    component.$html.find('.content-main').unbind('click', component.closeHandler);
+    component.$contentMain.unbind('click', component.closeHandler);
 };
 
-NavMainComponent.prototype.closeSubNavs = function() {
-
+NavMainComponent.prototype.closeSubNavs = function () {
     var component = this;
 
-    component.$html.find('.nav-secondary .nav-list').removeClass('is-active');
+    component.$navSecondary.find('.nav-list').removeClass('is-active');
+
+    if (component.$navQuaternary.hasClass('is-open')) {
+        component.$navQuaternary.removeClass('is-open');
+        component.$navQuaternary.find('.nav-list.is-active').removeClass('is-active');
+    } else {
+        component.$navTertiary.removeClass('is-open');
+    }
+};
+
+// Detect window height, adjust the number of items in the primary nav and check when to add "More" option
+NavMainComponent.prototype.adjustNavItems = function () {
+    var component = this,
+        availableHeight = component.$window.height(),
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true)),
+        moreIconHeight = 72, // Pre calculated height of the "More" nav item
```

Comment:
Ah okay, didn't realise this element was not attached to the DOM at this point, there is a method of temporarily adding it and querying the height then removing it - but I suggest we don't do that here. Might be worth adding to the comments _why_ we're hard coding it in that case.